core/src/ldml/ldml_transforms.cpp

```diff
@@ -11,6 +11,19 @@
 #include <string>
 #include "kmx/kmx_xstring.h"
 
+#if defined(HAVE_ICU4C)
+// TODO-LDML: Needed this for some compiler warnings
+#define U_FALLTHROUGH
+#include "unicode/uniset.h"
+#include "unicode/usetiter.h"
+#include "unicode/unistr.h"
+#include "unicode/regex.h"
+#include "unicode/utext.h"
+#include "unicode/utypes.h"
+#else
+#error icu4c is required for this code
+#endif
```

Comment:
can we precondition this instead of postcondition?
```suggestion
#if !defined(HAVE_ICU4C)
#error icu4c is required for this code
#endif

// TODO-LDML: Needed this for some compiler warnings
#define U_FALLTHROUGH
#include "unicode/uniset.h"
#include "unicode/usetiter.h"
#include "unicode/unistr.h"
#include "unicode/regex.h"
#include "unicode/utext.h"
#include "unicode/utypes.h"
```

Reply:
I actually had done exactly this just hadn't pushed it…great minds!